test-ts/test/erc20Precompile.test.ts

```diff
@@ -0,0 +1,78 @@
+import { JsonRpcProvider, Provider } from '@ethersproject/providers';
+import web3 from 'web3';
+import { expect, use } from 'chai';
+import { solidity } from 'ethereum-waffle';
+import { Contract, ContractFactory, Wallet } from 'ethers';
+import PrecompileCaller from '../artifacts/contracts/PrecompileCaller.sol/PrecompileCaller.json';
+
+use(solidity);
+
+describe('GA precompiles', () => {
+  let cennznetSigner: Wallet;
+  let cennznetProvider: Provider;
+  let precompileCaller: Contract;
+  let cennzToken: Contract;
+  const cennzTokenAddress = web3.utils.toChecksumAddress('0xCCCCCCCC00003e80000000000000000000000000');
+  const erc20Abi = [
+    'event Transfer(address indexed from, address indexed to, uint256 value)',
+    'event Approval(address indexed owner, address indexed spender, uint256 value)',
+    'function approve(address spender, uint256 amount) public returns (bool)',
+    'function allowance(address owner, address spender) public view returns (uint256)',
+    'function balanceOf(address who) public view returns (uint256)',
+    'function name() public view returns (string memory)',
+    'function symbol() public view returns (string memory)',
+    'function decimals() public view returns (uint8)',
+    'function transfer(address who, uint256 amount)',
+  ];
+
+  before(async () => {
+    cennznetProvider = new JsonRpcProvider('http://localhost:9933');
+    cennznetSigner = new Wallet('0xab3fdbe7eca16f38e9f5ee81b9e23d01ef251ba4ae19225783e5921a4a8c5564').connect(cennznetProvider); // 'development' seed
+    cennzToken = new Contract(cennzTokenAddress, erc20Abi, cennznetSigner);
+    console.log(`signer address=${cennznetSigner.address}`);
+  });
+
+  it('name, symbol, decimals', async () => {
+    expect(
+      await cennzToken.decimals()
+    ).to.equal(4);
+
+    expect(
+      await cennzToken.name()
+    ).to.equal("CENNZ");
+
+    expect(
+      await cennzToken.symbol()
+    ).to.equal("CENNZ");
+  });
+
+  it('approve and transferFrom', async () => {
+    let walletCENNZ = await cennzToken.balanceOf(cennznetSigner.address);
```

Comment:
does this given cennz generic asset balance in ethereum format?

Reply:
not sure what you mean by ethereum format? the value is in wei